### ISSUE TITLE
Installation via SPM Package Resolution Failed on leveldb 1.22.4 failing to clone third_party/googletest

### ISSUE THREAD

**Issue by MarceNN** [firebase/firebase-ios-sdk#12722]
### Description

Installing firebase -ios-sdk via SPM failed to resolve, with error messages like this:

**Failed to clone 'third_party/googletest' a second time, aborting**

### Reproducing the issue

Yes, it happens mostly when changing branches.
Reset Packages, Resolve Packages or Update to Latest Package Versions do not solve the issue.

### Firebase SDK Version

10.19.0

### Xcode Version

15.2

### Installation Method

Swift Package Manager

### Firebase Product(s)

Crashlytics

### Targeted Platforms

iOS

### Relevant Log Output

```shell
Checking out 1.22.4
Couldn’t update repository submodules:
    Submodule 'third_party/googletest' (https://github.com/google/googletest.git) registered for path 'third_party/googletest'
    Cloning into '/Users/.../SourcePackages/checkouts/leveldb/third_party/googletest'...
    error: RPC failed; curl 18 HTTP/2 stream 5 was reset
    error: 6164 bytes of body are still expected
    fetch-pack: unexpected disconnect while reading sideband packet
    fatal: early EOF
    fatal: fetch-pack: invalid index-pack output
    fatal: clone of 'https://github.com/google/googletest.git' into submodule path '/Users/.../SourcePackages/checkouts/leveldb/third_party/googletest' failed
    Failed to clone 'third_party/googletest'. Retry scheduled
    Cloning into '/Users/.../SourcePackages/checkouts/leveldb/third_party/googletest'...
    error: RPC failed; curl 18 HTTP/2 stream 5 was reset
    error: 4258 bytes of body are still expected
    fetch-pack: unexpected disconnect while reading sideband packet
    fatal: early EOF
    fatal: fetch-pack: invalid index-pack output
    fatal: clone of 'https://github.com/google/googletest.git' into submodule path '/Users/.../SourcePackages/checkouts/leveldb/third_party/googletest' failed
    Failed to clone 'third_party/googletest' a second time, aborting
Couldn’t update repository submodules:
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

{
  "pins" : [
    {
      "identity" : "abseil-cpp-binary",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/abseil-cpp-binary.git",
      "state" : {
        "revision" : "df308b8b46607675f2b9ec8e569109008f9155ce",
        "version" : "1.2022062300.1"
      }
    },
    {
      "identity" : "app-check",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/app-check.git",
      "state" : {
        "revision" : "3e464dad87dad2d29bb29a97836789bf0f8f67d2",
        "version" : "10.18.1"
      }
    },
    {
      "identity" : "firebase-ios-sdk",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/firebase-ios-sdk.git",
      "state" : {
        "revision" : "d9bcd141c3e4ad48a9500e6faeebb073f43cbcbd",
        "version" : "10.19.0"
      }
    },
    {
      "identity" : "googleappmeasurement",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleAppMeasurement.git",
      "state" : {
        "revision" : "6b332152355c372ace9966d8ee76ed191f97025e",
        "version" : "10.17.0"
      }
    },
    {
      "identity" : "googledatatransport",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleDataTransport.git",
      "state" : {
        "revision" : "a637d318ae7ae246b02d7305121275bc75ed5565",
        "version" : "9.4.0"
      }
    },
    {
      "identity" : "googleutilities",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleUtilities.git",
      "state" : {
        "revision" : "26c898aed8bed13b8a63057ee26500abbbcb8d55",
        "version" : "7.13.1"
      }
    },
    {
      "identity" : "grpc-binary",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/grpc-binary.git",
      "state" : {
        "revision" : "ea4cb5cc0c39c732b85386263116d2e2fdbbdc61",
        "version" : "1.49.2"
      }
    },
    {
      "identity" : "gtm-session-fetcher",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/gtm-session-fetcher.git",
      "state" : {
        "revision" : "9534039303015a84837090d20fa21cae6e5eadb6",
        "version" : "3.3.2"
      }
    },
    {
      "identity" : "interop-ios-for-google-sdks",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/interop-ios-for-google-sdks.git",
      "state" : {
        "revision" : "2d12673670417654f08f5f90fdd62926dc3a2648",
        "version" : "100.0.0"
      }
    },
    {
      "identity" : "leveldb",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/leveldb.git",
      "state" : {
        "revision" : "43aaef65e0c665daadf848761d560e446d350d3d",
        "version" : "1.22.4"
      }
    },
    {
      "identity" : "nanopb",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/firebase/nanopb.git",
      "state" : {
        "revision" : "819d0a2173aff699fb8c364b6fb906f7cdb1a692",
        "version" : "2.30909.0"
      }
    },
    {
      "identity" : "promises",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/promises.git",
      "state" : {
        "revision" : "540318ecedd63d883069ae7f1ed811a2df00b6ac",
        "version" : "2.4.0"
      }
    }
  ],
  "version" : 2
}


```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by paulb777** [comment]
We should remove the submodule from the forked https://github.com/firebase/leveldb.git, since it isn't used.

**Comment by MarceNN** [comment]
Hi, removing caches is not working any more.  Any other workaround ?

**Comment by ncooke3** [comment]
@MarceNN & @jay-jay-lama, I just published leveldb 1.22.5 which removed the submodule. Try updating your package workspace to pull in this new update.

**Comment by MarceNN** [comment]
It is working now. Thanks a lot.
